src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -316,7 +308,7 @@ impl<'tcx> RegionInferenceContext<'tcx> {
 
         while changed {
             changed = false;
-            for constraint in &*self.constraints {
+            for constraint in self.constraints.iter() {
                 if let Some(n) = result_set[constraint.sup] {
```

Comment:
Elsewhere, in `fn find_constraint_paths_between_regions_helper`, you have changed the direction of the `constraint` edge so that `'a: 'b` is represented as an edge `'a -> 'b` instead of `'b -> 'a`, which in effect meant that you replaced occurrences of `.sup` with `.sub`. Do you need to similarly revise the code in this method, swapping  `.sup` with `.sub`  and vice versa?

(In other words, I'm not sure I follow why the meaning of "'a may depend on 'b" seems to have changed in this commit. Unless the case in  `fn find_constraint_paths_between_regions_helper` was simply buggy?)

Reply:
Hmm, so I didn't touch this code, and it operates as it ever did. The problem here is that this code really ought to be removed -- it's an older variant that `find_constraint_paths_between_regions` has superceded, but it is still used in one or two places -- it'd probably be worth making an effort to remove it entirely, but I'm inclined to open an issue on that and defer it to another PR.